homeassistant/components/intellifire/__init__.py

```diff
@@ -31,79 +42,127 @@
-            password=entry.data[CONF_PASSWORD],
-        )
-    except (ConnectionError, ClientConnectionError) as err:
-        raise ConfigEntryNotReady from err
-    except LoginException as err:
-        raise ConfigEntryAuthFailed(err) from err
 
-    finally:
-        await ift_control.close()
 
-    # Extract API Key and User_ID from ift_control
-    # Eventually this will migrate to using IntellifireAPICloud
+async def _async_pseudo_migrate_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> ConfigEntry:
+    """Update configuration entry to latest VERSION 1 format.."""
+    new = {**config_entry.data}
+    # Rename Host to IP Address
+    new[CONF_IP_ADDRESS] = new.pop("host")
+
+    username = config_entry.data[CONF_USERNAME]
+    password = config_entry.data[CONF_PASSWORD]
 
-    if CONF_USER_ID not in entry.data or CONF_API_KEY not in entry.data:
-        LOGGER.info(
-            "Updating intellifire config entry for %s with api information",
-            entry.unique_id,
+    # Create a Cloud Interface
+    async with IntelliFireCloudInterface() as cloud_interface:
+        await cloud_interface.login_with_credentials(
+            username=username, password=password
         )
-        cloud_api = IntellifireAPICloud()
-        await cloud_api.login(
-            username=entry.data[CONF_USERNAME],
-            password=entry.data[CONF_PASSWORD],
+
+        # See if we can find the fireplace first by serial and then secondly by IP.
+
+        serial = config_entry.title.replace("Fireplace ", "")
+
+        # If serial matches the hex style pattern we'll assume its good
+        valid_serial = bool(re.match(r"^[0-9A-Fa-f]{32}$", serial))
+
+        new_data = (
+            cloud_interface.user_data.get_data_for_serial(serial)
+            if valid_serial
+            else None
         )
-        api_key = cloud_api.get_fireplace_api_key()
-        user_id = cloud_api.get_user_id()
-        # Update data entry
+        if not new_data:
+            new_data = cloud_interface.user_data.get_data_for_ip(new[CONF_IP_ADDRESS])
+
+    if not new_data:
+        raise ConfigEntryAuthFailed
+
+    # Find the correct fireplace
+    if new_data is not None:
+        new[CONF_API_KEY] = new_data.api_key
+        new[CONF_WEB_CLIENT_ID] = new_data.web_client_id
+        new[CONF_AUTH_COOKIE] = new_data.auth_cookie
+
+        new[CONF_IP_ADDRESS] = new_data.ip_address
+        new[CONF_SERIAL] = new_data.serial
+
+        config_entry.version = 1
         hass.config_entries.async_update_entry(
-            entry,
-            data={
-                **entry.data,
-                CONF_API_KEY: api_key,
-                CONF_USER_ID: user_id,
-            },
+            config_entry,
+            data=new,
+            options={CONF_READ_MODE: "local", CONF_CONTROL_MODE: "local"},
+            unique_id=serial,
         )
+        LOGGER.debug("Pseudo Migration %s successful", config_entry.version)
```

Comment:
is there a reason you're not doing a real migration?

Reply:
Yes -> its not necessary and @joostlek suggested if at all possible ignore doing a real migration

Reply:
We have minor migrations now, would prefer to have one of them (but only if its backwards compatible)

Reply:
It's definitely NOT backwards compatible at all... we're actually dropping one field the old version would need - so... I'm open to suggestions

Reply:
But do we have a reason to drop the field?

Reply:
🤷🏼 - Other than I don't love leaving junk around :) 

Manufacturer would prefer we move away from the old style of communications ...

Reply:
@joostlek  - do we have minor migration info somewhere in the docs ... I did a quick scan and couldn't find it. Is it like a version 1.1 or something?

Reply:
Can I close this out? Or do we need it open? (just rebased off latest branch)

Reply:
Minor version is indeed a thing, it's documented here: https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=minor_version#config-entry-migration

Please take a look at that, we should prefer that if it's possible.

"Backwards compatible" means the following steps should work:
1. User has a working intellifire config entry
2. User upgrade to a new version of Home Assistant Core which bumps the config entries minor version and does whatever changes is needed
3. User downgrades to the previous version of Home Assistant Core - at this point the config entry should work just like it did at step 1
4. User again upgrades to the new version of Home Assistant Core - note that at this point the migration logic may or may not run again depending on if the config entry data or options  were altered in step 3

Reply:
So I don't actually want to support downgrading - as the current method we are using is not inline with what the vendor (hearth and home) would like - as we are making extraneous cloud calls.

What do you propose in this situation? I just want to rewrite the way it works. I can make this as a breaking-change if need-be